### PR TITLE
[US Election] US election banner integration

### PR DESCRIPTION
**Pull request by chris-hinds**
Resolves #NUMBER

**Overall change:**
- Ads new container for the US2020 Election results banner
- Updates the frontPage to ensure it fetches the results oEmbed for `arabic, hindi, mundo, persian, portuguese, russian`
- Creates new feature toggle (currently on test only)
- Add a new route in the server for fetching fixture data of the results oEmbed

**Code changes:**
- [x] Create US Election Container
- [x] Include Banner fixture data
- [x] Add container to frontpage
- [x] Create feature toggle for election banner
- [x] Implement fetching of oEmbed data
- [x] Ensure data is only fetched for supported services
- [x] Write unit tests
- [x] Update snapshots

## This banner can/will be iterated on, this is just getting the 1st version available on test

- _A bullet point list of key code changes that have been made._
- _When describing code changes, try to communicate **how** and **why** you implemented something a specific way, not just **what** has changed._

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,3 +15,4 @@ simorgh.report.html
 simorgh.report.json
 cypress/integration/123PlaygroundForTests/*
 src/app/lib/config/services/*.test.js
+.env
```
Comment (chris-hinds) [inline .gitignore]: Not sure why this is showing as changed here...... it is like this on latest

File: scripts/bundleSize/bundleSizeConfig.js [hunk 2]
```diff
@@ -5,5 +5,5 @@ module.exports = {
   // below the smallest value in the build output; this avoids the
   // need for frequent changes as bundle sizes fluctuate.
   MIN_SIZE: 673,
-  MAX_SIZE: 851,
+  MAX_SIZE: 857,
```
Comment (chris-hinds) [inline scripts/bundleSize/bundleSizeConfig.js:8]: New size was 852. 857 follows our standard of +/- 5

File: src/app/containers/USElectionBanner/index.jsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { number, string, shape } from 'prop-types';
+import styled from 'styled-components';
```
Comment (karinathomasbbc) [inline src/app/containers/USElectionBanner/index.jsx:3]: 😭

File: src/app/routes/home/getInitialData/index.js [hunk 4]
```diff
@@ -16,14 +17,38 @@ const transformJson = pipe(
...
 
+const fetchElectionsOembed = async service => {
+  const usElectionOembedPath = `/${service}/election/us2020/results/oembed`;
+
+  try {
+    const { json, status } = await fetchPageData({
+      path: usElectionOembedPath,
+    });
+
+    if (json && status === 200) {
```
Comment (karinathomasbbc) [inline src/app/routes/home/getInitialData/index.js]: Could we use a static variable? https://github.com/bbc/simorgh/blob/latest/src/app/lib/statusCodes.const.js#L1

File: src/app/routes/home/getInitialData/index.js [hunk 5]
```diff
@@ -39,9 +64,13 @@ export default async ({ path: pathname, service, pageType, toggles }) => {
       pageData: {
         ...transformJson(json),
         radioSchedulePosition,
+        ...(pageHasUsElectionsBanner
+          ? await fetchElectionsOembed(service)
+          : null),
       },
     };
   } catch ({ message, status = getErrorStatusCode() }) {
+    console.log(message);
```
Comment (karinathomasbbc) [inline src/app/routes/home/getInitialData/index.js]: Oops

File: src/server/local/index.js [hunk 6]
```diff
@@ -176,6 +176,21 @@ export default server => {
       });
       sendDataFile(res, dataFilePath, next);
     })
+    .get(
```
Comment (karinathomasbbc) [inline src/server/local/index.js:179]: Could we add a test for this to https://github.com/bbc/simorgh/blob/latest/src/server/local/index.test.js

File: data/arabic/election/us2020/results/oembed.json [hunk 7]
```diff
@@ -0,0 +1,7 @@
...
...;-ms-transform:translateX(-50%) translateY(-50%);transform:translateX(-50%) translateY(-50%);}@media (min-width:20rem) and (max-width:37.4375rem){.css-nhr76b-StatesSoFar{font-size:1.125rem;line-height:1.5rem;}}@media (min-width:37.5rem){.css-nhr76b-StatesSoFar{font-size:1.125rem;line-height:1.5rem;}}@media (forced-colors:active){.css-nhr76b-StatesSoFar{border:0.0625rem solid transparent;}}@media (min-width:56.25rem){.css-nhr76b-StatesSoFar{top:calc(90px / 2);max-width:calc(100% - 2 * (4px + 90px));}}.css-nhr76b-StatesSoFar::before{content:' ';width:calc(2 * 8px);height:100%;position:absolute;top:0;right:100%;background:linear-gradient(to bottom left,transparent 50%,#ffffff 50%),linear-gradient(to top left,#222222 50%,#ffffff 50%);}.css-nhr76b-StatesSoFar::after{content:' ';width:calc(2 * 8px);height:100%;position:absolute;top:0;left:100%;background:linear-gradient(to bottom left,#ffffff 50%,transparent 50%),linear-gradient(to top left,#ffffff 50%,#222222 50%);}</style><p class=\"css-nhr76b-StatesSoFar e65i1318\">بعد أن أعلنت ولايات يبلغ عددها 50 من مجموع 50 النتائج فيها</p><script>console.log(\"Executing inline test script for US Election BBC banner.\");</script></div></div></div>"
```
Comment (simonsinclair) [inline data/arabic/election/us2020/results/oembed.json:6]: Ooh, look at that data tag. 😄

File: scripts/bundleSize/bundleSizeConfig.js (hunk 2 above)
Comment (simonsinclair) [reply scripts/bundleSize/bundleSizeConfig.js:8]: Did `MIN_SIZE` also change?

**Review by karinathomasbbc — APPROVED** [review]

File: src/app/containers/USElectionBanner/index.jsx [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+  GEL_GROUP_3_SCREEN_WIDTH_MAX,
+} from '@bbc/gel-foundations/breakpoints';
+
+// Utilities
+import useToggle from '#hooks/useToggle';
+
+// Components
+import FrontPageSection from '../../components/FrontPageSection';
+
+const createMarkup = oembed => {
```
Comment (simonsinclair) [inline src/app/containers/USElectionBanner/index.jsx:23]: create or `getMarkup`? 🤔

File: src/app/containers/USElectionBanner/index.jsx [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+  );
+};
+
+USElectionBanner.propTypes = {
+  oembed: shape({
+    version: string,
+    type: string,
+    width: number,
+    height: number,
+    html: string,
```
Comment (simonsinclair) [inline src/app/containers/USElectionBanner/index.jsx]: It's possible to set these properties as independently required if we'd like to, so even though `oembed` is optional, when it's defined `html` must be there. E.G.:
```
oembed: shape({
    version: string,
    type: string,
    width: number,
    height: number,
    html: string.isRequired,
}),
```

File: src/app/routes/home/getInitialData/index.js (hunk 5 above)
Comment (simonsinclair) [inline src/app/routes/home/getInitialData/index.js]: Is this to remain?

**Review by simonsinclair — APPROVED** [review]
Super work, Chris.

File: src/app/routes/home/getInitialData/index.js (hunk 5 above)
Comment (chris-hinds) [reply src/app/routes/home/getInitialData/index.js]: ahhhh, thought I cleaned all of these up 👍

File: src/app/containers/USElectionBanner/index.jsx (hunk 8 above)
Comment (chris-hinds) [reply src/app/containers/USElectionBanner/index.jsx:23]: I opted for create as is constructing the object required for `dangerouslySetInnerHTML` happy to change though.

File: src/app/routes/home/getInitialData/index.js (hunk 4 above)
Comment (chris-hinds) [reply src/app/routes/home/getInitialData/index.js]: done

File: src/app/routes/home/getInitialData/index.js (hunk 5 above)
Comment (chris-hinds) [reply src/app/routes/home/getInitialData/index.js]: done

File: src/server/local/index.js (hunk 6 above)
Comment (chris-hinds) [reply src/server/local/index.js:179]: done

File: src/app/containers/USElectionBanner/index.jsx (hunk 9 above)
Comment (chris-hinds) [reply src/app/containers/USElectionBanner/index.jsx]: done

File: src/app/routes/home/getInitialData/index.js (hunk 5 above)
Comment (chris-hinds) [reply src/app/routes/home/getInitialData/index.js]: done

**Pull request merged**